### PR TITLE
[tizen_app_control] Add static setAutoRestart method

### PR DESCRIPTION
**Pull request by swift-kim**
Add inhouse APIs related to the auto-restart feature as requested by an internal developer.

I don't have a Tizen 7.0 device that supports this feature. @hjhun Could you test the code?

Below is simple example code for testing. You can use the Send SMS and Pick image buttons to activate and deactivate the auto-restart setting.

```patch
diff --git a/packages/tizen_app_control/example/lib/main.dart b/packages/tizen_app_control/example/lib/main.dart
index ee06f16c..6d6fda84 100644
--- a/packages/tizen_app_control/example/lib/main.dart
+++ b/packages/tizen_app_control/example/lib/main.dart
@@ -87,9 +87,18 @@ class _MyAppState extends State<MyApp> {
     super.dispose();

     _localPort?.unregister();
+
+    AppControl.onAppControl.listen((ReceivedAppControl request) {
+      print('Received extra data: ${request.extraData}');
+    });
   }

   Future<void> _sendSms() async {
+    await AppControl.setAutoRestart(AppControl(
+      extraData: <String, dynamic>{'test_key': 'test_value'},
+    ));
+    return;
+
     final AppControl request = AppControl(
       operation: 'http://tizen.org/appcontrol/operation/share_text',
       uri: 'sms:',
@@ -109,6 +118,9 @@ class _MyAppState extends State<MyApp> {
   }

   Future<void> _pickImage() async {
+    await AppControl.unsetAutoRestart();
+    return;
+
     final AppControl request = AppControl(
       operation: 'http://tizen.org/appcontrol/operation/pick',
       mime: 'image/*',
```

### CODE REVIEW

**Comment by hjhun** [comment]
> 

Thank you first.
It works fine.
```
05-31 08:09:41.392+0900 W/AMD      (P 2227, T 2227): request_manager.cc: RequestHandler(465) > cyanra check. cmd(APP_SET_AUTO_RESTART:172), caller_pid(3830), req_id(1786)
05-31 08:09:41.393+0900 W/AMD_CYNARA_CORE (P 2227, T 2304): cynara_manager.cc: CynaraThread(358) > cynara_check() ++ User::Pkg::org.tizen.tizen_app_control_example|1685488163/proc/3830|5001|1786
05-31 08:09:41.393+0900 W/AMD_CYNARA_CORE (P 2227, T 2304): cynara_manager.cc: CynaraThread(362) > cynara_check() -- User::Pkg::org.tizen.tizen_app_control_example|1685488163/proc/3830|5001|1786
05-31 08:09:41.393+0900 W/AMD      (P 2227, T 2227): request_manager.cc: CynaraResponseCallback(403) > request has been allowed by cynara. cmd(APP_SET_AUTO_RESTART:172), caller(3830), req_id(1786)
05-31 08:09:41.393+0900 I/AMD      (P 2227, T 2227): request_manager.cc: DispatchRequest(278) > cmd(APP_SET_AUTO_RESTART:172), req_id(1786), caller_pid(3830), caller_uid(5001), clifd(37)
05-31 08:09:41.394+0900 W/AUL      (P 3830, T 3843): app_request.cc: SendSimply(131) > Request cmd(172:APP_SET_AUTO_RESTART): result(0)
05-31 08:09:48.823+0900 E/EFL      (P 3830, T 3830): ecore_wl2<3830> ../src/lib/ecore_wl2/ecore_wl2_input.c:681 _ecore_wl2_input_mouse_down_send() [press] window = 0x80a16080, time=162820 button=0
05-31 08:09:48.926+0900 E/EFL      (P 3830, T 3830): ecore_wl2<3830> ../src/lib/ecore_wl2/ecore_wl2_input.c:760 _ecore_wl2_input_mouse_up_send() [release] time=162925 button=0
05-31 08:09:48.933+0900 W/AUL      (P 3830, T 3843): app_request.cc: SendSimply(125) > Request cmd(172:APP_SET_AUTO_RESTART): target_uid(5001)
05-31 08:09:48.933+0900 D/AUL      (P 3830, T 3843): aul_sock.cc: SendAndReceive(227) > fd(83), cmd(172)
05-31 08:09:48.933+0900 W/AMD      (P 2227, T 2227): request_manager.cc: RequestHandler(465) > cyanra check. cmd(APP_SET_AUTO_RESTART:172), caller_pid(3830), req_id(1871)
05-31 08:09:48.934+0900 W/AMD_CYNARA_CORE (P 2227, T 2304): cynara_manager.cc: CynaraThread(358) > cynara_check() ++ User::Pkg::org.tizen.tizen_app_control_example|1685488163/proc/3830|5001|1871
05-31 08:09:48.934+0900 W/AMD_CYNARA_CORE (P 2227, T 2304): cynara_manager.cc: CynaraThread(362) > cynara_check() -- User::Pkg::org.tizen.tizen_app_control_example|1685488163/proc/3830|5001|1871
05-31 08:09:48.934+0900 W/AMD      (P 2227, T 2227): request_manager.cc: CynaraResponseCallback(403) > request has been allowed by cynara. cmd(APP_SET_AUTO_RESTART:172), caller(3830), req_id(1871)
05-31 08:09:48.934+0900 I/AMD      (P 2227, T 2227): request_manager.cc: DispatchRequest(278) > cmd(APP_SET_AUTO_RESTART:172), req_id(1871), caller_pid(3830), caller_uid(5001), clifd(37)
05-31 08:09:48.934+0900 W/AUL      (P 3830, T 3843): app_request.cc: SendSimply(131) > Request cmd(172:APP_SET_AUTO_RESTART): result(0)

root@localhost:/tmp# ps -ef | grep runner
owner     3830  2730  5 08:09 ?        00:00:02 /opt/usr/globalapps/org.tizen.tizen_app_control_example/bin/runner `zaybxcwdveuftgsh` __AUL_LOADER_ID__ JAAAAAEEAAASAAAAX19BVUxfTE9BREVSX0lEX18AAgAAADEA __AUL_TARGET_UID__ JQAAAAEEAAATAAAAX19BVUxfVEFSR0VUX1VJRF9fAAIAAAAwAA== __AUL_STARTTIME__ MAAAAAEEAAASAAAAX19BVUxfU1RBUlRUSU1FX18ADgAAADEzNy84Mjg0MDI2MTAA __APP_SVC_PKG_NAME__ SQAAAAEEAAAVAAAAX19BUFBfU1ZDX1BLR19OQU1FX18AJAAAAG9yZy50aXplbi50aXplbl9hcHBfY29udHJvbF9leGFtcGxlAA== __AUL_CALLER_PID__ KAAAAAEEAAATAAAAX19BVUxfQ0FMTEVSX1BJRF9fAAUAAAAzODI3AA== __AUL_CALLER_UID__ JQAAAAEEAAATAAAAX19BVUxfQ0FMTEVSX1VJRF9fAAIAAAAwAA== __AUL_HWACC__ IgAAAAEEAAAOAAAAX19BVUxfSFdBQ0NfXwAEAAAAVVNFAA== __AUL_ROOT_PATH__ WgAAAAEEAAASAAAAX19BVUxfUk9PVF9QQVRIX18AOAAAAC9vcHQvdXNyL2dsb2JhbGFwcHMvb3JnLnRpemVuLnRpemVuX2FwcF9jb250cm9sX2V4YW1wbGUA __AUL_INTERNAL_POOL__ LAAAAAEEAAAWAAAAX19BVUxfSU5URVJOQUxfUE9PTF9fAAYAAABmYWxzZQA= __AUL_API_VERSION__ KAAAAAEEAAAUAAAAX19BVUxfQVBJX1ZFUlNJT05fXwAEAAAANy4wAA== __AUL_IS_GLOBAL__ JwAAAAEEAAASAAAAX19BVUxfSVNfR0xPQkFMX18ABQAAAHRydWUA __AUL_INSTALLED_STORAGE__ MwAAAAEEAAAaAAAAX19BVUxfSU5TVEFMTEVEX1NUT1JBR0VfXwAJAAAAaW50ZXJuYWwA
root      4051  3547  0 08:10 pts/0    00:00:00 grep runner
root@localhost:/tmp# 
root@localhost:/tmp# kill -9 3830
root@localhost:/tmp# ps -ef | grep runner
owner     3029  2730  0 08:07 ?        00:00:01 /opt/usr/globalapps/org.tizen.tizen_app_control_example/bin/runner            
```

**Review by hjhun — APPROVED** [review]

**Comment by swift-kim** [comment]
@hjhun Does the app receive the app control data correctly on restart? That is, does the `print` statement in the sample code above print anything?

**Comment by hjhun** [comment]
> @hjhun Does the app receive the app control data correctly on restart? That is, does the `print` statement in the sample code above print anything?

I'm sorry. I did not check it. 
I've checked it. But, the extra data is empty.
At the time of setting auto restart, the bundle with empty extra data is being delivered.

**Comment by swift-kim** [comment]
What'd be the cause?

> At the time of setting auto restart, the bundle with empty extra data is being delivered.

Isn't the app control data supposed to be passed on app restart?

**Comment by hjhun** [comment]
> What'd be the cause?
> 
> > At the time of setting auto restart, the bundle with empty extra data is being delivered.
> 
> Isn't the app control data supposed to be passed on app restart?

I haven't checked in detail, but it seems that the extra data registered when creating the app control was not passed to the AUL API.

**Comment by swift-kim** [comment]
@hjhun Okay, I updated the code. You may check again.

**Comment by hjhun** [comment]
> @hjhun Okay, I updated the code. You may check again.

It works fine. Good.
```
05-31 15:08:11.029+0900 I/ConsoleMessage (P 3502, T 3825): flutter: Received extra data: {test_key: test_value}
```

**Review by hjhun — APPROVED** [review]

File: packages/tizen_app_control/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: tizen_app_control
 description: Tizen application control APIs. Used to launch apps on a Tizen device.
 homepage: https://github.com/flutter-tizen/plugins
 repository: https://github.com/flutter-tizen/plugins/tree/master/packages/tizen_app_control
-version: 0.2.2
+version: 0.2.3
```
Comment (swift-kim) [inline packages/tizen_app_control/pubspec.yaml:5]: Not increasing the minor version since the added API is a non-public API.

**Pull request merged**